storage/store.go

```diff
@@ -2435,21 +2440,23 @@ func (s *Store) processRaft() {
 				tickerStart := timeutil.Now()
 				s.processRaftMu.Lock()
 				s.mu.Lock()
-				for _, r := range s.mu.replicas {
-					if err := r.tick(); err != nil {
+				pending := make([]roachpb.RangeID, 0, len(s.mu.replicas))
```

Comment:
This is a big allocation if many of the ranges aren't pending. Perhaps we should not constantly reallocate pending and instead make it a store member variable which we set here as:

`pending = pending[:0]`